api/model.py

```diff
@@ -59,10 +59,29 @@ def post(self):
         # clean up from earlier runs
         if os.path.exists("/audio.wav"):
             os.remove("/audio.wav")
-
-        file = open("/audio.wav", "wb")
-        file.write(audio_data)
-        file.close()
+        
+        if os.path.exists("/audio.mp3"):
+            os.remove("/audio.mp3")
+        
+        #If the file is an mp3 file
+        #   Read into mp3.
+        #   Convert mp3 into wav using ffmpeg.
+        #Else read into wav file directly.
+        if('.mp3' in str(args['audio'])):
+            file = open("/audio.mp3", "wb")
+            file.write(audio_data)
+            file.close()
+            os.system("ffmpeg -i /audio.mp3 /audio.wav")
+            os.remove("/audio.mp3")
+        elif('.wav' in str(args['audio'])):
+            file = open("/audio.wav", "wb")
+            file.write(audio_data)
+            file.close()
+        else:
```

Comment:
This is going to return a `200` response for any invalid input. For example, passing in an image file will return a `200` OK response:

```
{
  "status": "error - invalid input file type",
  "predictions": []
}
```

What we want is to convert if we find an MP3, and if not assume it is a WAV file and proceed with prediction. We can wrap the `mw.predict` in a `try` - we should catch the right exception there so that in the case of invalid input we return a `400`. In case of another model error it would return `500`

Reply:
I had this logic in the original PR but changed it upon @ptitzler suggestion that we are to not assume that input is either `mp3` or `wav` only (bbbf9c5). Should I revert it to that case then?

Reply:
Well I guess what @ptitzler was saying is we should rather then detect invalid input here and fail with an explicit `400` response indicating a bad request.

That can be done either by being optimistic about input and going ahead with prediction, but then catching an appropriate error (if the input format is not correct). Alternatively, we need to do full input validation here (i.e. check that it is definitely a WAV or MP3 audio file). 

Relying on file extension is also perhaps brittle as then passing in a valid audio file without an extension will fail.

Reply:
I've gone the latter route in 1d5393a. It incorporates both the pre-validation that @ptitzler suggested and returning the right response code for invalid files. For now I am relying on the file extensions as the alternative would be to guess the file type in case of a missing extension and then trying to catch any errors downstream. 
I feel it is a fair assumption that the user will either upload files in one of the supported formats which is highlighted in the documentation or would correct it in case they get a `400` response in the first attempt as per current implementation.

Reply:
As long as invalid input returns HTTP code `400` I'm good with whatever approach you take.